packages/secrets/src/lib/adapters/aws-param-store.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  DeleteParameterCommand,
+  DescribeParametersCommand,
+  DescribeParametersCommandOutput,
+  GetParameterCommand,
+  GetParametersCommand,
+  ParameterNotFound,
+  PutParameterCommand,
+  SSMClient,
+} from '@aws-sdk/client-ssm';
+
+import type { SecretKey, SecretMap, SecretValue, SecretsVault } from '../types';
+
+export class AWSParameterStoreSecretsVault implements SecretsVault {
+  async deleteSecret(key: SecretKey) {
+    const client = new SSMClient();
```

Comment:
Would it make sense to move the `new SSMClient` to a constructor instead of creating it inside of each method here?

Reply:
I'll try and DRY it up.